src/bin/repl/mod.rs

```diff
@@ -114,26 +124,37 @@ impl Repl {
         let expr = analyze_expr(code)?;
         let expr_ty = expr.ctype.clone();
         let decl = wrap_expr(expr);
-        // FIXME: `ir` module is currently private.
         let module = ir::compile(module, vec![decl], false).0?;
 
         let mut jit = JIT::from(module);
         jit.finalize();
-        let execute_fun = jit
+        let fun = jit
             .get_compiled_function("execute")
             .expect("this is not good.");
 
         match expr_ty {
-            Type::Long(signed) => {
-                let result = unsafe {
-                    let execute: unsafe extern "C" fn() -> u64 = std::mem::transmute(execute_fun);
-                    execute()
-                };
-                match signed {
-                    true => println!("=> {}", result as i64),
-                    false => println!("=> {}", result),
-                }
-            }
+            Type::Short(signed) => execute!(fun, i16, |x| match signed {
+                true => println!("=> {}", x),
+                false => println!("=> {}", x as u16),
+            }),
+            Type::Int(signed) => execute!(fun, i32, |x| match signed {
+                true => println!("=> {}", x),
+                false => println!("=> {}", x as u32),
+            }),
+            Type::Long(signed) => execute!(fun, i64, |x| match signed {
+                true => println!("=> {}", x),
+                false => println!("=> {}", x as u64),
+            }),
+            Type::Float => execute!(fun, f32, |f| println!("=> {}", f)),
+            Type::Double => execute!(fun, f64, |f| println!("=> {}", f)),
+
+            Type::Char(_) => execute!(fun, char, |c| println!("=> {}", c)),
```

Comment:
This doesn't look quite right ... in Rust, `char` is a [unicode scalar character](https://doc.rust-lang.org/std/primitive.char.html) which is represented as 4 bytes. In C, it's an arbitrary byte, which could be invalid utf8 (`0xff`).
```suggestion
            Type::Char(_) => execute!(fun, u8, |c| println!("=> {}", if c.is_ascii() { c as char } else { `hex::encode(&[c]) })),
```
https://docs.rs/hex/0.4.2/hex/fn.encode.html